torch/distributed/_tensor/ops/pointwise_ops.py

```diff
@@ -376,11 +394,109 @@
 ]
 
 
+def pointwise_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> StrategyType:
+    max_shards_strategy_index = -1
+    max_shards = -1
+    # handle broadcasting
+    common_shape = torch.broadcast_shapes(
+        *[
+            arg.output_shape
+            for arg in op_schema.args_schema
+            if isinstance(arg, OpStrategy)
+        ]
+    )
+
+    if _is_inplace_op(op_schema.op):
+        # inplace op should follow the first arg strategy
+        followed_strategy = op_schema.args_schema[0]
+    elif _is_out_variant_op(op_schema.op):
+        # out variant op should follow the out kwarg strategy
+        followed_strategy = op_schema.kwargs_schema["out"]
+    else:
+        # normal pointwise op, we choose to follow the arg with
+        # the max shards in case operands needs reshard
+        for idx, arg_strategy in enumerate(op_schema.args_schema):
+            if not isinstance(arg_strategy, OpStrategy):
+                continue
+
+            arg_max_shards = arg_strategy.max_num_shards()
+            if arg_max_shards > max_shards:
+                max_shards_strategy_index = idx
+                max_shards = arg_max_shards
+
+        followed_strategy = op_schema.args_schema[max_shards_strategy_index]
+        assert isinstance(followed_strategy, OpStrategy)
+        follow_operand_dims_map = infer_broadcast_dims_map(
+            common_shape, followed_strategy.output_shape
+        )
+
+    assert isinstance(
+        followed_strategy, OpStrategy
+    ), f"no strategy to follow for {op_schema}!"
+    pointwise_strategy = OpStrategy([])
+
+    for placement_strategy in followed_strategy.strategies:
```

Comment:
Not a strong opinion here, but for better code readability, is it possible that we give the `followed_strategy` a better like, IIUC, this kind of implies how output (op result) can be distributed? For my learning purpose, why do we call it "follow"?

Reply:
And also, under what circumstances when there is multiple strategies for a followed_strategy?

Reply:
"follow" means all other args should follow this strategy. There might be cases there're multiple strategies for a graph mode case, for eager there's no case that there's multiple strategies